loadbalancer/algorithm.go

```diff
@@ -208,7 +226,9 @@ func (r *random) Apply(ctx *routing.LBContext) routing.LBEndpoint {
 		return ctx.Route.LBEndpoints[i]
 	}
 
-	return withFadeIn(r.rand, ctx, r.notFadingIndexes, i, r)
+	i = withFadeIn(r.rand, ctx, r.notFadingIndexes, i, r)
+	i = withHealthCheck(r.rand, ctx, r.notFadingIndexes, i, r)
```

Comment:
I wonder if it would not be better to actually have a new LB algorithm instead of modify random. Since you avoid unhealthy node it is not exactly random anymore.

Reply:
This same thing could be applied to round robin also. Instead of a whole LB algorithm this could be just an option to both algorithms like and configurable through some flag or parameter.

Reply:
We need to have a strategy pattern somewhere.
I think what we want is:
- probability based drops (or better re-schedule), similar to admissionControl reject calculation to allow small amount of traffic to broken endpoints to observe PHC (passive health check)
- exponential backoff